addons/test_mail_full/tests/test_portal.py

```diff
@@ -240,19 +237,6 @@ def setUpClass(cls):
                         sort=True),
              parsed_url[4]))
 
-    def assert_URL(self, url, expected_path, expected_fragment_params=None, expected_query=None):
-        """Asserts that the URL has the expected path and if set, the expected fragment parameters and query."""
-        parsed_url = url_parse(url)
-        fragment_params = url_decode(parsed_url.fragment)
-        self.assertEqual(parsed_url.path, expected_path)
-        if expected_fragment_params:
-            for key, expected_value in expected_fragment_params.items():
-                self.assertEqual(fragment_params.get(key), expected_value,
-                                 f'Expected: "{key}={expected_value}" (for path: {expected_path})')
-        if expected_query:
-            self.assertEqual(expected_query, parsed_url.query,
-                             f'Expected: query="{expected_query}" (for path: {expected_path})')
-
```

Comment:
Not sure why you need to remove this !

Reply:
`assert_URL` function is parsing the url and taking its fragment(part of url after `#`) and query(part of url after `?`) and mathing it with expected_fragment_params and expected_query. And also is matching the parsed_url.path with expected_path.

Currently in this file we are getting value of `res.url` in test cases like this - 
`http://127.0.0.1:8069/my?subpath=action-mail.action_discuss` which is passed to `assert_URL`.

So in this case we will never get fragment so we will never be able to match it with expected_fragment_params.
You can refer to [Runbot Build](https://runbot.odoo.com/runbot/build/65968276) for more reference.

Kindly give me your suggestions on this. Thanks!!

Reply:
Ok, I forgot that the redirections are modified also.
Thanks

Reply:
You can still use the function and modify the call from `expected_fragment_params ` to `expected_query `
The URL part after a `?` it's called a query. So in this case you should check if the exepected_query is equal to `subpath=action-mail.action_discuss`